src/components/ec-btn/ec-btn.vue

```diff
@@ -0,0 +1,169 @@
+<template>
+  <component
+    :is="componentTag"
+    v-bind="componentProps"
+    class="ec-btn"
+    :class="buttonClasses"
+  >
+    <ec-icon
+      v-if="icon"
+      class="ec-btn__icon"
+      :class="{
+        'ec-btn--icon-with-spacing': this.$slots.default && icon,
+        'ec-btn__content-invisible' : loading && !this.$slots['text-loader'],
+        'ec-btn__content-remove' : loading && this.$slots['text-loader'],
+      }"
+      :name="icon"
+      :size="22"
+    />
+
+    <span
+      v-if="loading && this.$slots['text-loader']"
+      data-test="ec-btn__text-loader-slot"
+    >
+      <slot name="text-loader" />
+    </span>
+
+    <span
+      v-else-if="this.$slots.default"
+      data-test="ec-btn__default-slot"
+      :class="{'ec-btn__content-invisible' : loading }"
+    >
+      <slot />
+    </span>
+
+    <ec-loading
+      v-if="loading && !this.$slots['text-loader']"
+      class="ec-btn__spinner"
+      show
+      :size="24"
+    />
+
+  </component>
+</template>
+
+<script>
+import EcIcon from '../ec-icon/ec-icon.vue';
+import EcLoading from '../ec-loading/ec-loading.vue';
+
+export default {
+  name: 'EcBtn',
+  components: {
+    EcIcon,
+    EcLoading,
+  },
+  props: {
+    size: {
+      type: String,
+      default: 'md',
+      validator(value) {
+        return ['sm', 'md'].includes(value);
+      },
+    },
+    href: {
+      type: String,
+    },
+    to: {
+      type: String,
+    },
+    tag: {
+      type: String,
+    },
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+    icon: {
+      type: String,
+    },
+    rounded: {
+      type: Boolean,
+      default: false,
+    },
+    outline: {
+      type: Boolean,
+      default: false,
+    },
+    fullWidth: {
+      type: Boolean,
+      default: false,
+    },
+    type: {
```

Comment:
@matallo @jaimeperezpedra. We were discussing this with @mcibique. Shall we rename from "type" to "category"? Type seems bit vague :/